adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/CoapAdapterProperties.java

```diff
@@ -26,13 +26,47 @@
      * The default regular expression to split the identity into authority and tenant.
      */
     public static final String DEFAULT_ID_SPLIT_REGEX = "@";
+    /**
+     * The default number of threads for the connector.
+     */
+    public static final int DEFAULT_CONNECTOR_THREADS = 2;
+    /**
+     * The default number of threads for the coap protocol stack.
+     */
+    public static final int DEFAULT_COAP_THREADS = 2;
+    /**
+     * The default number of threads for the dtls connector.
+     */
+    public static final int DEFAULT_DTLS_THREADS = 32;
+    /**
+     * The default timeout in milliseconds for DTLS retransmissions.
+     */
+    public static final int DEFAULT_DTLS_RETRANSMISSION_TIMEOUT = 2000;
+    /**
+     * The default exchange lifetime in milliseconds.
+     * 
+     * According <a href= "https://tools.ietf.org/html/rfc7252#page-30"> RFC 7252 - 4.8. Transmission Parameters</a> the
+     * default value is 247 seconds. Such a large time requires also a huge amount of heap. That time includes a
+     * processing time of 100s and retransmissions of CON messages. Therefore a practical value could be much smaller.
+     */
+    public static final int DEFAULT_EXCHANGE_LIFETIME = 247000;
+    /**
+     * The default for message offloading. When messages are kept for deduplication, some parts of the messages are not
+     * longer required and could be release earlier. That helps to reduce the heap consumption. Value
+     * {@value #DEFAULT_MESSAGE_OFFLOADING}.
+     */
+    public static final boolean DEFAULT_MESSAGE_OFFLOADING = true;
 
     private String idSplitRegex = DEFAULT_ID_SPLIT_REGEX;
     private String networkConfig = null;
     private String secureNetworkConfig = null;
     private String insecureNetworkConfig = null;
-    private int connectorThreads = 1;
-    private int coapThreads = 2;
+    private int connectorThreads = DEFAULT_CONNECTOR_THREADS;
+    private int coapThreads = DEFAULT_COAP_THREADS;
+    private int dtlsThreads = DEFAULT_DTLS_THREADS;
+    private int dtlsRetransmissionTimeout = DEFAULT_DTLS_RETRANSMISSION_TIMEOUT;
+    private int exchangeLifetime = DEFAULT_EXCHANGE_LIFETIME;
+    private boolean messageOffloading = DEFAULT_MESSAGE_OFFLOADING;
```

Comment:
please rename this to *messageOffloadingEnabled*

Reply:
done